### DISCUSSION TITLE
similar word prediction for custom ner model

### DISCUSSION THREAD

**Discussion by NaziaToma** [explosion/spaCy#9026] [Help: Coding & Implementations] [answered]
I have trained model for custom ner. My code & output is -

```
doc = nlp("shewrapara, Road number 2, House 761")
print([(ent.text, ent.label_) for ent in doc.ents])
```
Output-
SUB_AREA shewrapara
ROAD_NUMBER Road number 2
HOUSE House 761

But if I write the sub-area name wrong then-
```
doc = nlp("sheorapara, Road number 2, House 761")
print([(ent.text, ent.label_) for ent in doc.ents])
```
result -
ROAD_NUMBER sheorapara, Road number 2
HOUSE House 761

Here the distance between shewrapara & sheorapara is 1 as we can see. So is there any way to calculate this similarity distance in spacy and show the right label for slightly changed words?

**Comment by polm** [comment, marked as answer]
No, there is no reasonable way for the model to take edit distance into account directly. It doesn't have an internal list of "all strings that are `SUB_AREA`", so it can't check edit distance to them.

If you have enough training data, the model should be able to learn things like:

- `SUB_AREA` is likely to begin with `she`
- `SUB_AREA` is likely to end with `ara`
- `SUB_AREA` is likely to be followed by `, Road number`

Those may not all be true, but if you have enough data the model will be able to pick up on the correct patterns. 

What you can do is use [data augmentation](https://spacy.io/usage/training#data-augmentation) during training to make the model better at dealing with small changes like this. You can randomly replace letters, so that in the best case your model might even see `sheopara` in training. Replacement can be totally random or more principled (maybe you mainly replace vowels with other vowels or something). 

As another point, if you have a list of all valid values for `SUB_AREA`, then you could use an EntityRuler for that, or create a custom component to check individual words against the list and check for edit distance.
